mne_rsa/rsa.py

```diff
@@ -156,30 +156,44 @@ def rsa_gen(rdm_data_gen, rdm_model, metric="spearman", ignore_nan=False):
         return_array = False
         rdm_model = [_ensure_condensed(rdm_model, "rdm_model")]
 
+
     if ignore_nan:
+        if cache_ranks and (rsa_metric == "spearman"):
+            raise ValueError("ignore_nan and cache_ranks is not yet supported together")
         masks = [~np.isnan(rdm) for rdm in rdm_model]
     else:
         masks = [slice(None)] * len(rdm_model)
 
+    if cache_ranks and (rsa_metric == "spearman"):
+        # Precompute ranks for Spearman
+        rdm_model = [stats.rankdata(rdm) for rdm in rdm_model]
+
     for rdm_data in rdm_data_gen:
         rdm_data = _ensure_condensed(rdm_data, "rdm_data")
         if ignore_nan:
             data_mask = ~np.isnan(rdm_data)
             masks = [m & data_mask for m in masks]
-        rsa_vals = _rsa_single_rdm(rdm_data, rdm_model, metric, masks)
+        rsa_vals = _rsa_single_rdm(rdm_data, rdm_model, metric, masks, cache_ranks)
         if return_array:
             yield np.asarray(rsa_vals)
         else:
             yield rsa_vals[0]
 
 
-def _rsa_single_rdm(rdm_data, rdm_model, metric, masks):
+def _rsa_single_rdm(rdm_data, rdm_model, metric, masks, cache_ranks):
     """Compute RSA between a single data RDM and one or more model RDMs."""
     if metric == "spearman":
-        rsa_vals = [
-            stats.spearmanr(rdm_data[mask], rdm_model_[mask])[0]
-            for rdm_model_, mask in zip(rdm_model, masks)
-        ]
+        if cache_ranks:
+            rdm_data = stats.rankdata(rdm_data)
+            rsa_vals = [
+                np.corrcoef(rdm_data, rdm_model_[mask])[0,1]
+                for rdm_model_, mask in zip(rdm_model, masks)
+            ]
+        else:
+            rsa_vals = [
+                stats.spearmanr(rdm_data[mask], rdm_model_[mask])[0]
+                for rdm_model_, mask in zip(rdm_model, masks)
+            ]
     elif metric == "pearson":
         rsa_vals = [
             stats.pearsonr(rdm_data[mask], rdm_model_[mask])[0]
```

Comment:
this should also probably become `np.corrcoef(a, b)[0, 1]` as it is indeed faster than `stats.pearsonr`.